libcontainer/nsenter/nsexec.c

```diff
@@ -560,6 +613,11 @@ void nsexec(void)
 	if (ensure_cloned_binary() < 0)
 		bail("could not ensure we are a cloned binary");
 
+	/* Get the log pipe */
+	logfd = logpipe();
```

Comment:
I'd suggest making `logfd` be set globally inside `logpipe()` and rename it `setup_logpipe()` -- since you never pass `logfd` to another function.

Reply:
This should also be done earlier so that `bail` can be redefined as using the logging pipe.

Reply:
Done